include/stdexec/execution.hpp

```diff
@@ -5589,7 +5535,15 @@ namespace stdexec {
           (!__tag_invocable_with_completion_scheduler<
             sync_wait_t, set_value_t, _Sender>) &&
           tag_invocable<sync_wait_t, _Sender>
-      tag_invoke_result_t<sync_wait_t, _Sender>
+      // TODO - for reasons I cannot figure out, the move_only_t test in
+      // test/nvexec/then.cpp appears to instantiate this version of
+      // operator(), even though the complation scheduler overload above is
+      // what actually is called. This leads the compilation to fail when
+      // computing tag_invoke_result_t in the return type.
+      // Using decltype(auto) avoids the tag_invoke_result_t expansion.
+      // Why does the compiler not respect the requires clause and avoid
+      // expanding the tag_invoke_result_t?
```

Comment:
Well, compilers are buggy. They really _shouldn't_ be trying to instantiate the return type when the requirements aren't met, but that wasn't the case until the resolution of a core issue (whose number I can't remember).

But. You should never get a hard error from a use of `tag_invoke_result_t` in a SFINAE context. If that's happening to you, then there is a bug elsewhere, and using `decltype(auto)` here is just hiding it.

Reply:
It might be CWG 2369, which I don't think clang implements yet.

Reply:
Yes, that's the one.

Reply:
Ah well, as luck would have it, removing the `same_as<get_attrs_t> _Tag` bit from then.cuh ended up fixing this. Still need to dig deeper as to why though..